tests/document/indexes.py

```diff
@@ -792,6 +792,27 @@ class User(Document):
 
         User.drop_collection()
 
+    def test_unique_and_primary_create(self):
+        """Create a new record with a duplicate primary key
+        throws an exception
+        """
+
```

Comment:
Thank you very much!
Just one question regarding the changed you requested - I have based my code on the previously existing function, test_unique_and_primary, and that function might require the same changes to be done:
- getting rid of inline function in favor of "with"
- removing the empty line after the comment
- removing the redundant drop_collection call

Do you want me to do the above changes on that function (which is not mine) also? It is no problem at all, I just wanted to be sure :)